### ISSUE TITLE
Test should fail when no conformance URI is found in the API model

### ISSUE THREAD

**Issue by heidivanparys** [opengeospatial/ets-ogcapi-features10#116]
Class Conformance contains the following:

``` java
    @DataProvider(name = "conformanceUris")
    public Object[][] conformanceUris( ITestContext testContext ) {
        OpenApi3 apiModel = (OpenApi3) testContext.getSuite().getAttribute( API_MODEL.getName() );
        URI iut = (URI) testContext.getSuite().getAttribute( IUT.getName() );
        List<TestPoint> testPoints = retrieveTestPointsForConformance( apiModel, iut );
        Object[][] testPointsData = new Object[testPoints.size()][];
        int i = 0;
        for ( TestPoint testPoint : testPoints ) {
            testPointsData[i++] = new Object[] { testPoint };
        }
        return testPointsData;
    }
```

When testPointsData is an empty array, the test should fail.

Example of an API where no conformance class is found: the root is here defined as https://my.url.com and the path as /abc/xyz/conformance, whereas the root should be https://my.url.com/abc/xyz and the path /conformance .

Unless that would be a valid API? But then the test should be updated to actually find https://my.url.com/abc/xyz/conformance from the API definition.

``` json
{
  "openapi": "3.0.3",
  "info": {
    "title": "Title",
    "description": "Description",
    "termsOfService": "https://link.to.terms",
    "contact": {
      "name": "Name",
      "url": "https://link.to.contact"
    },
    "version": "1.0"
  },
  "servers": [
    {
      "url": "https://my.url.com",
      "description": "Description"
    }
  ],
  "paths": {
    "/abc/xyz/conformance": {
      "get": {
        "tags": [
          "Capabilities"
        ],
        "summary": "Get the list of conformance classes that this service implements",
        "operationId": "getTheConformance",
        "responses": {
          "200": {
            "description": "Successful Operation",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Conformance"
                },
                "example": {
                  "conformsTo": [
                    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core",
                    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30",
                    "http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson"
                  ]
                }
              }
            }
          }
        }
      }
    }
  }
}

```

**Comment by heidivanparys** [comment]
> Example of an API where no conformance class is found: the root is here defined as https://my.url.com and the path as /abc/xyz/conformance, whereas the root should be https://my.url.com/abc/xyz and the path /conformance .
> 
> Unless that would be a valid API?

That is actually a valid API, see https://github.com/opengeospatial/ogcapi-features/issues/427.

However, the test should still fail when no conformance URI is found in the API model.

Abstract test 7 (/conf/core/conformance-op) reads:

> 1. Construct a path for each "conformance" link on the landing page as well as for the {root}/conformance path.
> 2. Issue an HTTP GET request on each path

Test method `Conformance.validateConformanceOperationAndResponse(TestPoint)` is supposed to test abstract test 7, but it has a test point as parameter. The test points are retrieved in method `Conformance.conformanceUris(ITestContext)`. So the API model must describe a resource URL that is `{iut}/conformance`, otherwise the API is not conformant to requirement /req/core/conformance-op.

In addition, a path for each "conformance" link on the landing page should be constructed and a HTTP GET request issued, I am not sure that that is done in the test? Or to avoid redundancy in calling the API: find the conformance links from the landing page and check that they are the same URL as `{iut}/conformance`.

**Comment by dstenger** [comment]
Thank you for reporting. We will do further investigation.

**Comment by dstenger** [comment]
@keshav-nangare Can you please implement that method `validateConformanceOperationAndResponse` throws a test failure when data provider `conformanceUris` returns an empty array?

@heidivanparys I agree that the test suite is only retrieving `{iut}/conformance`. However, in my understanding, `{root}/conformance` should be equal to (each) conformance link on the landing page.
Thus, it is hard for me to interpret the aim of the abstract test. E.g. what happens if the content of multiple links is different? Is it even allowed to provide multiple conformance links?
That is the reason, the implementation is currently kept simple here.
Maybe, you have further ideas?

**Comment by keshav-nangare** [comment]
@dstenger

If the DataProvider is empty then the test `validateConformanceOperationAndResponse ` is not triggered by DataProvider.
I have updated the test which will throw the exception if `conformanceUris` dataProvider is empty but it shows as skipped in the HTML report. 

For more reference the test result after changes:

![image](https://user-images.githubusercontent.com/14732650/93113468-da0de780-f6d6-11ea-8bf0-13d345e2b53e.png)

Fixed with #120 PR.

**Comment by keshav-nangare** [comment]
@dstenger 

As per our discussion, I have updated the test and now it throws an exception.

![image](https://user-images.githubusercontent.com/14732650/93481641-66f5b280-f91c-11ea-9d46-0dbb5171a9bc.png)

The changes are pushed to the same PR.
